### PR TITLE
Ar 41/en tant que developpeur je souhaite pouvoir creer un compte sur lapplication depuis un endpoint

### PR DESCRIPTION
**Pull request by romainpanno**
* **Checklist before requesting a review**
- [x] The commits messages follows our guidelines
- [x] I have performed a self-review of my code
- [x] I have documented my code
- [x] Do we need to implement unit tests?

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of changes does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [x] Feature request
- [ ] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **Describe your changes**
add /register route

* **Issue ticket number and link**
AR-41?
https://bravmm.atlassian.net/browse/AR-41?atlOrigin=eyJpIjoiYjA1MWJlZTFlYmI3NGY5Y2FhOGE3NGFmM2RhY2FkODIiLCJwIjoiaiJ9

## Summary by CodeRabbit

- **New Features**
  - Introduced user registration and login endpoints with enhanced security measures.
  - Implemented JSON Web Token (JWT) generation for secure user authentication.

- **Enhancements**
  - Improved user password security with bcrypt hashing.
  - Added input validation for user registration and login to ensure data integrity.

- **Bug Fixes**
  - Resolved issues with user creation and login flow to enhance reliability.

- **Documentation**
  - Updated API documentation to reflect new registration and login endpoints.

- **Refactor**
  - Replaced deprecated middleware with `body-parser` for improved JSON request handling.
  - Streamlined database synchronization process for better startup performance.

- **Chores**
  - Environment variable checks implemented to ensure necessary configurations are set.

### CODE REVIEW

File: back/src/route/user.route.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
...
+import * as userController from '../controller/user.controller';
 
-const userController = require('../controller/user.controller');
 
-router.post('/create', userController.createUser);
+router.post('/register', userController.register);
+router.post('/login', userController.login);
 
-module.exports = router;
+export default router;
```
Comment (Beafowl-Pull) [inline back/src/route/user.route.ts:9]:
```suggestion
export default router;

```

File: back/src/models/index.ts [hunk 2]
```diff
@@ -33,4 +32,4 @@ db.reactions = Reaction.initialize(sequelize);
 db.actionData = ActionData.initialize(sequelize);
 db.reactionData = ReactionData.initialize(sequelize);
 
-export default db;
+export default db;
```
Comment (Beafowl-Pull) [inline back/src/models/index.ts:35]:
```suggestion
export default db;

```

**Review by Beafowl-Pull — CHANGES_REQUESTED** [review]

**Review by Beafowl-Pull — APPROVED** [review]
RAS

File: back/package.json [hunk 3]
```diff
@@ -11,16 +11,21 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "bcrypt": "^5.1.1",
     "cors": "^2.8.5",
     "dotenv": "^16.3.1",
     "express": "^4.18.2",
+    "jsonwebtoken": "^9.0.2",
     "mysql2": "^3.6.5",
```
Comment (badidu09) [inline back/package.json:19]: remove it if possible

**Review by badidu09 — CHANGES_REQUESTED** [review]
check comment

File: back/package.json (hunk 3 above)
Comment (romainpanno) [reply back/package.json:19]: It's to automatically parse json body

Comment (romainpanno) [reply back/package.json:19]: Like to generate jwt, and I need it to import and not require

Comment (badidu09) [reply back/package.json:19]: I miss click i want to click on the line under : mysql2

Comment (Beafowl-Pull) [inline back/package.json:19]:
```suggestion
```

**Review by Beafowl-Pull — CHANGES_REQUESTED** [review]

**Review by Beafowl-Pull — DISMISSED** [review]
RAS

**Review by badidu09 — DISMISSED** [review]

**Review by Beafowl-Pull — DISMISSED** [review]

**Review by badidu09 — APPROVED** [review]

**Review by Beafowl-Pull — APPROVED** [review]

**Review by badidu09 — DISMISSED** [review]

**Review by Beafowl-Pull — APPROVED** [review]

**Review by badidu09 — DISMISSED** [review]

**Review by Vivelis — APPROVED** [review]

**Review by Vivelis — APPROVED** [review]

**Review by Vivelis — APPROVED** [review]

**Review by Beafowl-Pull — APPROVED** [review]

**Pull request merged**